core/src/main/java/ai/timefold/solver/core/impl/bavet/common/AbstractFlattenLastNode.java

```diff
@@ -169,4 +130,36 @@ public Propagator getPropagator() {
         return propagationQueue;
     }
 
+    private record FlattenLastMapEntry<FlattenedItem_, OutTuple_>(FlattenedItem_ value, MutableInt newCount,
+            List<OutTuple_> outTupleList) {
+        FlattenLastMapEntry(FlattenedItem_ value) {
+            this(value, new MutableInt(), new ArrayList<>());
+        }
+
+        void add(Supplier<OutTuple_> outTupleSupplier,
+                Consumer<OutTuple_> insertConsumer,
+                Consumer<OutTuple_> updateConsumer) {
+            var listIndex = newCount.intValue();
+            newCount.increment();
+            if (newCount.intValue() > outTupleList.size()) {
+                var inserted = outTupleSupplier.get();
+                outTupleList.add(inserted);
+                insertConsumer.accept(inserted);
+            } else {
+                updateConsumer.accept(outTupleList.get(listIndex));
+            }
+        }
+
+        void retract(Consumer<OutTuple_> retractedConsumer) {
+            for (var i = newCount.intValue(); i < outTupleList.size(); i++) {
+                retractedConsumer.accept(outTupleList.get(i));
```

Comment:
If the other consumers are called `insertConsumer` and `updateConsumer`, shouldn't this be called `retractConsumer`? The other two do not use past tense.